From 470f2c68f0a1143e215f94d6076bb802e2011072 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Tue, 29 Jan 2013 20:11:20 +0200
Subject: [PATCH] Configure upscaling also on refresh adjustment

If playback configures screen refresh rate while keeping same resolution, xrandr is called but configuring refresh adjustment isn't reached.
This patch adds this support

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 xbmc/windowing/X11/XRandR.cpp |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/xbmc/windowing/X11/XRandR.cpp b/xbmc/windowing/X11/XRandR.cpp
index 1c7feb7..9ddecb9 100644
--- a/xbmc/windowing/X11/XRandR.cpp
+++ b/xbmc/windowing/X11/XRandR.cpp
@@ -291,15 +291,15 @@ bool CXRandR::SetMode(XOutput output, XMode mode)
     return false;
   CLog::Log(LOGINFO, "XRANDR: %s", cmd);
   int status = system(cmd);
+#ifdef TARGET_MARVELL_DOVE
+  ChangeGraphicsScaler ();
+#endif
   if (status == -1)
     return false;
 
   if (WEXITSTATUS(status) != 0)
     return false;
 
-#ifdef TARGET_MARVELL_DOVE
-  ChangeGraphicsScaler ();
-#endif
   return true;
 }
 
-- 
1.7.9.5

